pilot/pkg/serviceregistry/serviceentry/controller.go

```diff
@@ -345,11 +345,15 @@ func (s *Controller) workloadEntryHandler(old, curr config.Config, event model.E
 	// update eds cache only
 	s.edsUpdate(allInstances, false)
 
+	// TODO: if len(configsUpdated) == 0 we're firing a Forced push for compatibility because no configs have been updated,
+	// we should actually skip the push.
 	pushReq := &model.PushRequest{
 		Full:           true,
 		ConfigsUpdated: configsUpdated,
 		Reason:         model.NewReasonStats(model.EndpointUpdate),
+		Forced:         len(configsUpdated) == 0,
```

Comment:
Not sure why cannot skip push if = 0

Reply:
Previously it wasn't skipped, I didn't want to introduce unnecessary changes to this PR and also I think it's not really safe to currently skip pushing

Reply:
Yeah, i guess you must met something wrong, just wanna to know what is it